src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -277,6 +289,26 @@ class CORE_EXPORT QgsSymbolLegendNode : public QgsLayerTreeModelLegendNode
      */
     void setTextOnSymbolTextFormat( const QgsTextFormat &format ) { mTextOnSymbolTextFormat = format; }
 
+    /**
+     * Label of the symbol, user defined label will be used, otherwise will default to the label made by QGIS.
+     * \since QGIS 3.10
+     */
+    QString symbolLabel() const;
+
+    /**
+     * Create an expressionContextScope containing symbol related variables
+     * \since QGIS 3.10
+     */
+    QgsExpressionContextScope *createSymbolScope() const;
+
+    /**
+     * Evaluates  and returns the text label of the current node
+     * \param context extra QgsExpressionContext to use for evaluating the expression
+     * \param label text to evaluate instead of the layer layertree string
+     * \since QGIS 3.10
+     */
+    QString evaluateLabel( const QgsExpressionContext &context = QgsExpressionContext(), const QString &label = QString() );
```

Comment:
Is this `label` or `labelExpression`?

Reply:
maybe labelExpression would be more appropriate, you are right.

Reply:
In hindsight I would leave as is. The reason being that this is the label that can be provided as is by the user and that the member function of the layer currently named labelExpression is used in the code. This would only make things far more confusing.

Reply:
Are you ok with my explanation?

Reply:
Yes, sounds good :+1: